Open Data Hub Operator provisioning doesn't work

**Describe the bug**
Intersmash testsuite integration tests are failing when using community deliverables on OpenShift. Specifically the failure is about `OpenDataHubOpenShiftOperatorProvisionerTest`, which logs report the following traces:

```
...
12:05:10 [ERROR] org.jboss.intersmash.testsuite.provision.openshift.OpenDataHubOpenShiftOperatorProvisionerTest.testMinimalDataScienceCluster  Time elapsed: 316.923 s  <<< ERROR!
12:05:10 io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
12:05:10 	at org.jboss.intersmash.testsuite.provision.openshift.OpenDataHubOpenShiftOperatorProvisionerTest.verifyMinimalDataScienceCluster(OpenDataHubOpenShiftOperatorProvisionerTest.java:149)
12:05:10 	at org.jboss.intersmash.testsuite.provision.openshift.OpenDataHubOpenShiftOperatorProvisionerTest.testMinimalDataScienceCluster(OpenDataHubOpenShiftOperatorProvisionerTest.java:139)
12:05:10 Caused by: io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
12:05:10 Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: 
12:05:10 Unrecognized field "feastoperator" (class io.opendatahub.datasciencecluster.v1.datascienceclusterspec.Components), not marked as ignorable (11 known properties: "codeflare", "dashboard", "modelmeshserving", "datasciencepipelines", "trustyai", "ray", "kueue", "workbenches", "trainingoperator", "kserve", "modelregistry"])
12:05:10  at [Source: (BufferedInputStream); line: 1, column: 818] (through reference chain: io.opendatahub.datasciencecluster.v1.DataScienceCluster["spec"]->io.opendatahub.datasciencecluster.v1.DataScienceClusterSpec["components"]->io.opendatahub.datasciencecluster.v1.datascienceclusterspec.Components["feastoperator"])
...
```

**To Reproduce**
`mvn clean test -pl testsuite/integration-tests -Pts.openshift -pl testsuite/integration-tests -Dtest=OpenDataHubOpenShiftOperatorProvisionerTest`

**Expected behavior**
The test should pass

**Screenshots**
N/A

**Additional context**
This seems to be related with outdated/incompatible CRDs definitions, which should be updated by including the latest manifests.

--- a/provisioners/src/main/resources/crds/datasciencecluster.opendatahub.io_datascienceclusters.yaml
+++ b/provisioners/src/main/resources/crds/datasciencecluster.opendatahub.io_datascienceclusters.yaml
@@ -1,9 +1,11 @@
+# See https://raw.githubusercontent.com/opendatahub-io/opendatahub-operator/refs/tags/v2.25.0/bundle/manifests/datasciencecluster.opendatahub.io_datascienceclusters.yaml
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.16.1
+  creationTimestamp: null
   name: datascienceclusters.datasciencecluster.opendatahub.io
 spec:
   group: datasciencecluster.opendatahub.io
@@ -143,6 +145,52 @@ spec:
                     description: |-
                       DataSciencePipeline component configuration.
                       Requires OpenShift Pipelines Operator to be installed before enable component
+                    properties:
+                      devFlags:
+                        description: Add developer fields
+                        properties:
+                          manifests:
+                            description: List of custom manifests for the given component
+                            items:
+                              properties:
+                                contextDir:
+                                  default: manifests
+                                  description: contextDir is the relative path to
+                                    the folder containing manifests in a repository,
+                                    default value "manifests"
+                                  type: string
+                                sourcePath:
+                                  default: ""
+                                  description: 'sourcePath is the subpath within contextDir
+                                    where kustomize builds start. Examples include
+                                    any sub-folder or path: `base`, `overlays/dev`,
+                                    `default`, `odh` etc.'
+                                  type: string
+                                uri:
+                                  default: ""
+                                  description: uri is the URI point to a git repo
+                                    with tag/branch. e.g.  https://github.com/org/repo/tarball/<tag/branch>
+                                  type: string
+                              type: object
+                            type: array
+                        type: object
+                      managementState:
+                        description: |-
+                          Set to one of the following values:
+
+                          - "Managed" : the operator is actively managing the component and trying to keep it active.
+                                        It will only upgrade the component if it is safe to do so
+
+                          - "Removed" : the operator is actively managing the component and will not install it,
+                                        or if it is installed, the operator will try to remove it
+                        enum:
+                        - Managed
+                        - Removed
+                        pattern: ^(Managed|Unmanaged|Force|Removed)$
+                        type: string
+                    type: object
+                  feastoperator:
+                    description: Feast Operator component configuration.
                     properties:
                       devFlags:
                         description: Add developer fields
@@ -676,6 +724,24 @@ spec:
                         - Removed
                         pattern: ^(Managed|Unmanaged|Force|Removed)$
                         type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
                     type: object
                   dashboard:
                     description: Dashboard component status.
@@ -714,9 +780,27 @@ spec:
                         - Removed
                         pattern: ^(Managed|Unmanaged|Force|Removed)$
                         type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
                     type: object
-                  kserve:
-                    description: Kserve component status.
+                  feastoperator:
+                    description: Feast Operator component status.
                     properties:
                       managementState:
                         description: |-
@@ -730,6 +814,68 @@ spec:
                         enum:
                         - Managed
                         - Removed
+                        pattern: ^(Managed|Unmanaged|Force|Removed)$
+                        type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
+                    type: object
+                  kserve:
+                    description: Kserve component status.
+                    properties:
+                      defaultDeploymentMode:
+                        description: |-
+                          DefaultDeploymentMode is the value of the defaultDeploymentMode field
+                          as read from the "deploy" JSON in the inferenceservice-config ConfigMap
+                        type: string
+                      managementState:
+                        description: |-
+                          Set to one of the following values:
+
+                          - "Managed" : the operator is actively managing the component and trying to keep it active.
+                                        It will only upgrade the component if it is safe to do so
+
+                          - "Removed" : the operator is actively managing the component and will not install it,
+                                        or if it is installed, the operator will try to remove it
+                        enum:
+                        - Managed
+                        - Removed
+                        pattern: ^(Managed|Unmanaged|Force|Removed)$
+                        type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
+                      serverlessMode:
                         pattern: ^(Managed|Unmanaged|Force|Removed)$
                         type: string
                     type: object
@@ -750,6 +896,24 @@ spec:
                         - Removed
                         pattern: ^(Managed|Unmanaged|Force|Removed)$
                         type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
                     type: object
                   modelmeshserving:
                     description: ModelMeshServing component status.
@@ -768,6 +932,24 @@ spec:
                         - Removed
                         pattern: ^(Managed|Unmanaged|Force|Removed)$
                         type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
                     type: object
                   modelregistry:
                     description: ModelRegistry component status.
@@ -788,6 +970,24 @@ spec:
                         type: string
                       registriesNamespace:
                         type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
                     type: object
                   ray:
                     description: Ray component status.
@@ -806,6 +1006,24 @@ spec:
                         - Removed
                         pattern: ^(Managed|Unmanaged|Force|Removed)$
                         type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
                     type: object
                   trainingoperator:
                     description: Training Operator component status.
@@ -824,6 +1042,24 @@ spec:
                         - Removed
                         pattern: ^(Managed|Unmanaged|Force|Removed)$
                         type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
                     type: object
                   trustyai:
                     description: TrustyAI component status.
@@ -842,6 +1078,24 @@ spec:
                         - Removed
                         pattern: ^(Managed|Unmanaged|Force|Removed)$
                         type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
                     type: object
                   workbenches:
                     description: Workbenches component status.
@@ -860,6 +1114,24 @@ spec:
                         - Removed
                         pattern: ^(Managed|Unmanaged|Force|Removed)$
                         type: string
+                      releases:
+                        items:
+                          description: ComponentRelease represents the detailed status
+                            of a component release.
+                          properties:
+                            name:
+                              type: string
+                            repoUrl:
+                              type: string
+                            version:
+                              type: string
+                          required:
+                          - name
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - name
+                        x-kubernetes-list-type: map
                     type: object
                 type: object
               conditions:
@@ -970,3 +1242,9 @@ spec:
     storage: true
     subresources:
       status: {}
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: null
+  storedVersions: null

--- a/provisioners/src/main/resources/crds/dscinitialization.opendatahub.io_dscinitializations.yaml
+++ b/provisioners/src/main/resources/crds/dscinitialization.opendatahub.io_dscinitializations.yaml
@@ -1,9 +1,11 @@
+# See https://raw.githubusercontent.com/opendatahub-io/opendatahub-operator/refs/tags/v2.25.0/bundle/manifests/dscinitialization.opendatahub.io_dscinitializations.yamlapiVersion: apiextensions.k8s.io/v1
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.16.1
+  creationTimestamp: null
   name: dscinitializations.dscinitialization.opendatahub.io
 spec:
   group: dscinitialization.opendatahub.io
@@ -82,7 +84,9 @@ spec:
                     - default
                     type: string
                   manifestsUri:
-                    description: Custom manifests uri for odh-manifests
+                    description: |-
+                      ## DEPRECATED ## : ManifestsUri set on DSCI is not maintained.
+                      Custom manifests uri for odh-manifests
                     type: string
                 type: object
               monitoring:
@@ -208,8 +212,6 @@ spec:
                 - customCABundle
                 - managementState
                 type: object
-            required:
-            - applicationsNamespace
             type: object
           status:
             description: DSCInitializationStatus defines the observed state of DSCInitialization.
@@ -313,3 +315,9 @@ spec:
     storage: true
     subresources:
       status: {}
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: null
+  storedVersions: null

--- a/provisioners/src/main/resources/crds/features.opendatahub.io_featuretrackers.yaml
+++ b/provisioners/src/main/resources/crds/features.opendatahub.io_featuretrackers.yaml
@@ -1,9 +1,11 @@
+# See https://raw.githubusercontent.com/opendatahub-io/opendatahub-operator/refs/tags/v2.25.0/bundle/manifests/features.opendatahub.io_featuretrackers.yamlapiVersion: apiextensions.k8s.io/v1
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.16.1
+  creationTimestamp: null
   name: featuretrackers.features.opendatahub.io
 spec:
   group: features.opendatahub.io
@@ -99,3 +101,9 @@ spec:
     storage: true
     subresources:
       status: {}
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: null
+  storedVersions: null

--- a/provisioners/src/main/resources/crds/services.platform.opendatahub.io_monitorings.yaml
+++ b/provisioners/src/main/resources/crds/services.platform.opendatahub.io_monitorings.yaml
@@ -1,9 +1,11 @@
+# See https://raw.githubusercontent.com/opendatahub-io/opendatahub-operator/refs/tags/v2.25.0/bundle/manifests/services.platform.opendatahub.io_monitorings.yaml
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.16.1
+  creationTimestamp: null
   name: monitorings.services.platform.opendatahub.io
 spec:
   group: services.platform.opendatahub.io
@@ -66,39 +68,41 @@ spec:
             properties:
               conditions:
                 items:
-                  description: Condition contains details for one aspect of the current
-                    state of this API Resource.
                   properties:
+                    lastHeartbeatTime:
+                      description: |-
+                        The last time we got an update on a given condition, this should not be set and is
+                        present only for backward compatibility reasons
+                      format: date-time
+                      type: string
                     lastTransitionTime:
                       description: |-
                         lastTransitionTime is the last time the condition transitioned from one status to another.
-                        This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable.
+                        This should be when the underlying condition changed.
+                        If that is not known, then using the time when the API field changed is acceptable.
                       format: date-time
                       type: string
                     message:
-                      description: |-
-                        message is a human readable message indicating details about the transition.
-                        This may be an empty string.
-                      maxLength: 32768
+                      description: message is a human-readable message indicating
+                        details about the transition.
                       type: string
                     observedGeneration:
                       description: |-
                         observedGeneration represents the .metadata.generation that the condition was set based upon.
-                        For instance, if .metadata.generation is currently 12, but the .status.conditions[x].observedGeneration is 9, the condition is out of date
-                        with respect to the current state of the instance.
+                        For instance, if .metadata.generation is currently 12, but the .status.conditions[x].observedGeneration
+                        is 9, the condition is out of date with respect to the current state of the instance.
                       format: int64
                       minimum: 0
                       type: integer
                     reason:
                       description: |-
                         reason contains a programmatic identifier indicating the reason for the condition's last transition.
-                        Producers of specific condition types may define expected values and meanings for this field,
-                        and whether the values are considered a guaranteed API.
                         The value should be a CamelCase string.
-                        This field may not be empty.
-                      maxLength: 1024
-                      minLength: 1
-                      pattern: ^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$
+                      type: string
+                    severity:
+                      description: |-
+                        Severity with which to treat failures of this type of condition.
+                        When this is not specified, it defaults to Error.
                       type: string
                     status:
                       description: status of the condition, one of True, False, Unknown.
@@ -113,17 +117,13 @@ spec:
                       pattern: ^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$
                       type: string
                   required:
-                  - lastTransitionTime
-                  - message
-                  - reason
                   - status
                   - type
                   type: object
                 type: array
-                x-kubernetes-list-map-keys:
-                - type
-                x-kubernetes-list-type: map
+                x-kubernetes-list-type: atomic
               observedGeneration:
+                description: The generation observed by the resource controller.
                 format: int64
                 type: integer
               phase:
@@ -139,3 +139,9 @@ spec:
     storage: true
     subresources:
       status: {}
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: null
+  storedVersions: null

